### DISCUSSION TITLE
Incompatible input shape

### DISCUSSION THREAD

**Discussion by edmundngu** [apache/incubator-mxnet#20234] [Q&A]
May I know why i'm getting this error? 
I believe that all the layers in my model are set correctly because I can train it without any problem. 
However, the problem below occur when I test my model.
FYI, the problem arise when the model concat an upsampling layer with a conv layer.
Any solution?

Thanks to your help.

```
MXNetError: MXNetError: Error in operator RF_core_concat32: [05:38:05] ../src/operator/nn/concat.cc:67: Check failed: shape_assign(&(*in_shape)[i], dshape): Incompatible input shape: expected [1,-1,32,47], got [1,256,32,48]
```

**Comment by szha** [comment]
Looks like there's a mismatch in the size of the last dimension in the infer_shape of `RF_core_concat32`.
```
Incompatible input shape: expected [1,-1,32,47], got [1,256,32,48]
```
The operator doesn't seem to be from mxnet. Are you using custom op?

**Reply by edmundngu** [reply]
Thanks for your help @szha .
I found the solution for this problem.
I use the function below to crop the shape in order to match them up.

``` python
mx.symbol.Crop
```
